### PR TITLE
Marketing: Add "Sales" column in Campaigns card

### LINKED ISSUE #43248: [Enhancement]: Marketing: Add "Sales" column in Campaigns card
### Describe the solution you'd like

Currently, in the Campaigns card in the Marketing > Overview page (`/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`), we have a "Cost" column. See screenshot below:

image

In this issue, we want to add a new "Sales" column. Each marketing channel should be able to report their own sales figure (similar to the existing cost figure), and the figure should be displayed in the Campaigns card.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ecgan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/43248.

In this PR, we add a new Sales column in the Campaigns card in the Marketing page.

When there are no values for the cost or sales columns, it will display a dash "-".

The sales column implementation is based on the existing implementation for cost column.

Screenshot:

image

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Note: to test the changes, you may need to use some test data in some marketing channel extensions. The following is an example of how to do that using Google Listings and Ads:

1. Modify the line https://github.com/woocommerce/google-listings-and-ads/blob/72543805c6a24191c33c5cd0177d66b07537b18b/src/MultichannelMarketing/GLAChannel.php#L75 to the following:

```php
		if ( apply_filters( 'woocommerce_gla_enable_mcm', true ) === true ) {
```

2. Modify the `get_campaigns` function in https://github.com/woocommerce/google-listings-and-ads/blob/72543805c6a24191c33c5cd0177d66b07537b18b/src/MultichannelMarketing/GLAChannel.php#L178-L206, so that it returns some sample marketing campaigns from MailPoet and Google Listings and Ads:

```php
	public function get_campaigns(): array {
		return [
			new MarketingCampaign(
				'1',
				$this->campaign_types['google-ads'],
				'Campaign 1 - no cost, no sales',
				admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit' ),
			),
			new MarketingCampaign(
				'2',
				$this->campaign_types['google-ads'],
				'Campaign 2 - no cost, with sales',
				admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit' ),
				null,
				new Price( '30', 'USD' ),
			),
			new MarketingCampaign(
				'3',
				$this->campaign_types['google-ads'],
				'Campaign 3 - with cost, no sales',
				admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit' ),
				new Price( '30', 'USD' ),
			),
			new MarketingCampaign(
				'4',
				$this->campaign_types['google-ads'],
				'Campaign 4 - with cost, with sales',
				admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit' ),
				new Price( '5,000', 'USD' ),
				new Price( '6,000', 'USD' ),
			),
			new MarketingCampaign(
				'5',
				$this->campaign_types['google-ads'],
				'Campaign 5',
				admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit' ),
				new Price( '5,000', 'USD' ),
				new Price( '6,000', 'USD' ),
			),
			new MarketingCampaign(
				'6',
				$this->campaign_types['google-ads'],
				'Campaign 6',
				admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit' ),
				new Price( '6,888.5', 'USD' ),
			),
		];
	}
```

Test Steps:

1. Go to the WooCommerce > Marketing page: `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`. 
2. You should see Campaigns card like the screenshot above.
    - You should see Cost and Sales column.
    - If there are no values for cost or sales, a dash "-" would be displayed.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/marketing/hooks/useCampaigns.ts [hunk 1]
```diff
@@ -48,13 +48,18 @@ export const useCampaigns = ( page = 1, perPage = 5 ): UseCampaignsType => {
 
 				const cost = campaign.cost
 					? `${ campaign.cost.currency } ${ campaign.cost.value }`
-					: '';
+					: '-';
+
+				const sales = campaign.sales
+					? `${ campaign.sales.currency } ${ campaign.sales.value }`
```
Comment (ibndawood) [inline plugins/woocommerce-admin/client/marketing/hooks/useCampaigns.ts:54]: ❓ Is it part of the design to not format this price value as per user settings in WooCommerce?

**Review by ibndawood — APPROVED** [review]
Thank you, @ecgan. I followed the test instructions and can confirm that the changes work as described.

I've raised a question which is not related to this PR.

File: plugins/woocommerce/src/Admin/API/MarketingCampaigns.php [hunk 2]
```diff
@@ -168,55 +175,73 @@ public function prepare_item_for_response( $item, $request ) {
...
-				'cost'       => [
+				),
+				'cost'       => array(
 					'description' => __( 'Cost of the marketing campaign.', 'woocommerce' ),
-					'context'     => [ 'view' ],
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+					'type'        => 'object',
+					'properties'  => array(
+						'value'    => array(
```
Comment (budzanowski) [inline plugins/woocommerce/src/Admin/API/MarketingCampaigns.php:213]: Indentation?

File: plugins/woocommerce/src/Admin/API/MarketingCampaigns.php [hunk 3]
```diff
@@ -168,55 +175,73 @@ public function prepare_item_for_response( $item, $request ) {
...
+				),
+				'sales'      => array(
+					'description' => __( 'Sales of the marketing campaign.', 'woocommerce' ),
+					'context'     => array( 'view' ),
 					'readonly'    => true,
 					'type'        => 'object',
-					'properties'  => [
-						'value'    => [
+					'properties'  => array(
+						'value'    => array(
```
Comment (budzanowski) [inline plugins/woocommerce/src/Admin/API/MarketingCampaigns.php:231]: Indentation? Not sure but looks off.

Comment (ibndawood) [reply plugins/woocommerce/src/Admin/API/MarketingCampaigns.php:231]: It does align with the `currency` index.

File: plugins/woocommerce/src/Admin/API/MarketingCampaigns.php (hunk 2 above)
Comment (ibndawood) [reply plugins/woocommerce/src/Admin/API/MarketingCampaigns.php:213]: It does align with the `currency` index.

File: plugins/woocommerce-admin/client/marketing/hooks/useCampaigns.ts (hunk 1 above)
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/hooks/useCampaigns.ts:54]: This has been raised recently, and I will create a follow up issue and work on this. 🙏

Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/hooks/useCampaigns.ts:54]: I have created issue https://github.com/woocommerce/woocommerce/issues/43797 for this.

**Pull request merged**